lib/operations/execute_operation.js

```diff
@@ -65,25 +65,25 @@ function executeOperation(topology, operation, callback) {
   }
 
   const makeExecuteCallback = (resolve, reject) =>
-    function executeCallback(err, result) {
+    function executeCallback(err, result, server) {
       if (session && session.owner === owner) {
         session.endSession(() => {
           if (operation.session === session) {
             operation.clearSession();
           }
           if (err) return reject(err);
-          resolve(result);
+          resolve(result, server);
```

Comment:
you cannot resolve a promise with more than one value, and for uniformity we should not be doing this with callbacks either.

Reply:
All ears for suggested alternative approach to informing a cursor what server was selected for the operation.

Reply:
Can it be part of the `AggregateOperation`? You wind up passing the server to the agg op [here](https://github.com/mongodb/node-mongodb-native/pull/2075/files/3a1c0221cd2109eb107022707524bad0632baf47#diff-e9be4765aec928f35fb2dec80e92e19bR177).

Reply:
done